### ISSUE TITLE
failed to reproduce the results

### ISSUE THREAD

**Issue by qizhust** [IMNearth/Curriculum-Learning-For-VLN#3]
Thanks for the code! I failed to reproduce the results in Table 3. 1) the results from follower+naive CL and follower+self-paced Cl are the same; 2) the reproduced results are generally worse than the reported ones, and sometimes conflict; 3) the results do not change with random seeds. Do you have any suggestions?

|                  | val_seen |          |          |          |          | val_unseen |          |          |          |          |
|------------------|:--------:|:--------:|:--------:|:--------:|:--------:|:----------:|:--------:|:--------:|:--------:|:--------:|
| reported         |      NE↓ |       SR |      OSR |      SPL |      CLS |        NE↓ |       SR |      OSR |      SPL |      CLS |
| follower         | 4.85     | 52.3     | 65.3     | 44.3     | 58.2     | 7.12       | 28.6     | 40.9     | 20.3     | 35.0     |
|   +naïve CL      | 5.03     | 48.6     | 62.0     | 40.4     | 55.9     | 7.13       | 31.1     | 42.8     | 21.2     | 34.3     |
|   +self-paced CL | **4.23** | **58.7** | **69.2** | **51.1** | **63.3** | **6.69**   | **32.2** | **44.2** | **24.5** | **38.9** |
| reproduced       |          |          |          |          |          |            |          |          |          |          |
| follower         | **4.40** | 56.4     | 68.2     | 47.6     | **60.9** | 7.23       | 26.2     | 38.7     | 17.8     | 31.9     |
|   +naïve CL      | 4.45     | 55.2     | 67.4     | 44.4     | 56.4     | **6.80**   | **30.0** | **43.1** | **20.2** | **34.3** |
|   +self-paced CL | 4.44     | **58.2** | **69.3** | **48.3** | 58.8     | 7.05       | 28.3     | 39.2     | 19.8     | 34.1     |
|                  |          |          |          |          |          |            |          |          |          |          |
| reported         |          |          |          |          |          |            |          |          |          |          |
| self-monitoring  | 4.27     | 58.4     | 67.0     | 51.9     | 64.1     | 6.42       | 38.4     | 48.1     | 28.3     | 41.5     |
|   +naïve CL      | **4.08** | **61.0** | **69.8** | **54.6** | 64.9     | 6.30       | 40.0     | 51.7     | 28.6     | 41.1     |
|   +self-paced CL | 4.19     | 58.8     | 68.2     | 53.3     | **65.4** | **5.98**   | **41.0** | **52.4** | **30.8** | **43.9** |
| reproduced       |          |          |          |          |          |            |          |          |          |          |
| self-monitoring  | 4.17     | 58.2     | 67.4     | 52.9     | 65.3     | 6.45       | 35.6     | 46.0     | 27.2     | 41.9     |
|   +naïve CL      | **3.98** | **61.9** | **70.2** | **55.5** | 66.2     | **6.20**   | 39.9     | 50.2     | 29.5     | 41.7     |
|   +self-paced CL | 4.02     | 59.9     | 69.3     | 54.8     | **66.6** | 6.28       | **40.1** | **51.2** | **31.2** | **44.7** |
|                  |          |          |          |          |          |            |          |          |          |          |
| reported         |          |          |          |          |          |            |          |          |          |          |
| envdrop          | 4.55     | 57.7     | **65.6** | 54.4     | 67.2     | 5.92       | 45.7     | 54.2     | 41.8     | 57.0     |
|   +naïve CL      | 4.49     | 57.8     | 63.1     | **54.8** | 67.2     | 5.93       | 44.3     | 50.5     | 41.3     | 57.6     |
|   +self-paced CL | **4.42** | **58.1** | **65.6** | **54.8** | **67.4** | **5.48**   | **47.6** | **54.3** | **44.1** | **59.1** |
| reproduced       |          |          |          |          |          |            |          |          |          |          |
| envdrop          | 4.57     | 57.3     | **65.7** | 54.0     | **67.1** | **5.86**   | **45.6** | 53.6     | **41.8** | 57.3     |
|   +naïve CL      | 4.89     | 56.4     | 63.5     | 53.3     | 65.7     | 5.91       | 43.9     | 51.9     | 40.4     | **57.4** |
|   +self-paced CL | **4.44** | **57.8** | 65.4     | **54.5** | 67.0     | 5.94       | 44.7     | **53.7** | 41.5     | 57.3     |

**Comment by IMNearth** [comment]
Could you please give me more details about your training settings and local machine information? 

Moreover, it is strange that the results do not change with random seeds. Our observation is the results will fluctuate with random seed. We have reported the results from repeating experiments in Table 7.

**Comment by qizhust** [comment]
Thanks for the reply!

I build a singularity image to run the code, with ubuntu16.04 torch==1.7.1+cu101 torchvision==0.8.2+cu101 torchaudio==0.7.2 python3.8.5.

**Comment by IMNearth** [comment]
Hi Qi Zheng,

My suggestion is that you should switch your python and torch version as the same in paper. However, I will also try your settings. I will contact you when the results come out.

With regard to your reproduced experiments, although the results from follower and self-monitoring are slightly lower than that in paper, they do show that curriculum learning is useful. For envdrop agent, maybe you should try again. Since I have repeated the experiments many times in my machine, the performance changes with the random seed, but exhibit a similar trend that SPCL > NCL > ML.

**Comment by qizhust** [comment]
Thanks! I will try the setting and let you know in case of any changes.

**Comment by qizhust** [comment]
Hi, I use the same version as suggested. The result of follower and self-monitoring is consistent with that in the paper, however, the result envdrop trained by SPCL is worse than the baseline. Following results are on the val_unseen split.

|                 | NE             | SR              | OSR             | SPL             |
|-----------------|----------------|-----------------|-----------------|-----------------|
| Follower        | 6.98(0.14)     | 29.71(1.68)     | 40.75(1.45)     | 20.66(1.46)     |
|     +naïve CL   | 7.03(0.07)     | 29.95(0.84)     | **42.92**(1.34) | 20.02(0.99)     |
|     +self-paced | **6.75**(0.11) | **32.08**(0.77) | 40.39(1.16)     | **23.83**(0.86) |
| reproduced      |                |                 |                 |                 |
| Follower        | 7.18(0.12)     | 27.1(1.64)      | 38.1(1.83)      | 19.5±1.42       |
|   +naïve CL     | 6.98(0.11)     | 30.0(0.73)      | **42.1**(1.44)  | 20.9±0.79       |
|   +self-paced   | **6.81**(0.11) | **30.9**(1.54)  | 42.0(1.25)      | **23.1**±1.47   |
|                 |                |                 |                 |                 |
| Self-monitor    | 6.35(0.07)     | 38.22(1.32)     | 48.65(2.01)     | 29.15(1.73)     |
|     +naïve CL   | 6.35(0.06)     | 39.83(0.43)     | **50.44**(1.60) | 29.77(1.30)     |
|     +self-paced | **6.18**(0.14) | **40.22**(1.33) | 50.05(1.96)     | **32.12**(1.16) |
| reproduced      |                |                 |                 |                 |
| Self-monitor    | 6.47(0.12)     | 36.0(1.26)      | 47.3(2.04)      | 27.5(1.33)      |
|   +naïve CL     | 6.45(0.04)     | 37.1(1.76)      | 48.1(1.95)      | 28.2(1.67)      |
|   +self-paced   | **6.41**(0.11) | **38.3**(1.84)  | **49.6**(1.95)  | **29.4**(1.74)  |
|                 |                |                 |                 |                 |
| envdrop         | 5.79(0.09)     | 45.44(0.28)     | 54.11(0.89)     | 41.84(0.30)     |
|     +naïve CL   | 5.91(0.09)     | 44.66(0.84)     | 51.91(0.97)     | 41.46(0.72)     |
|     +self-paced | **5.71**(0.15) | **46.11**(0.89) | **54.13**(1.21) | **42.52**(0.98) |
| reproduced      |                |                 |                 |                 |
| envdrop         | **5.87**(0.09) | **44.6**(1.18)  | **52.6**(1.13)  | **41.3**(1.11)  |
|   +naïve CL     | 6.05(0.09)     | 43.9(0.84)      | 51.1(1.06)      | 40.7(0.67)      |
|   +self-paced   | 5.98(0.05)     | 43.9(0.68)      | 52.0(0.74)      | 40.7(0.50)      |
